plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/Base85Codec.java

```diff
@@ -59,42 +59,31 @@ public final class Base85Codec
 
     private Base85Codec() {}
 
-    public static ByteBuffer decodeBlocks(String encoded)
+    public static ByteBuffer decode(String encoded)
     {
         char[] input = encoded.toCharArray();
         checkArgument(input.length % 5 == 0, "Input should be 5 character aligned");
         ByteBuffer buffer = ByteBuffer.allocate(input.length / 5 * 4);
 
-        // A mechanism to detect invalid characters in the input while decoding, that only has a
-        // single conditional at the very end, instead of branching for every character.
-        class InputCharDecoder
-        {
-            int canary;
-
-            long decodeInputChar(int i)
-            {
-                char c = input[i];
-                canary |= c; // non-ascii char has bits outside of ASCII_BITMASK
-                byte b = DECODE_MAP[c & ASCII_BITMASK];
-                canary |= b; // invalid char maps to -1, which has bits outside ASCII_BITMASK
-                return b;
-            }
-        }
-
         int inputIndex = 0;
-        InputCharDecoder inputCharDecoder = new InputCharDecoder();
-        while (buffer.hasRemaining()) {
+        while (inputIndex < input.length) {
             int sum = 0;
-            sum += inputCharDecoder.decodeInputChar(inputIndex) * BASE_4TH_POWER;
-            sum += inputCharDecoder.decodeInputChar(inputIndex + 1) * BASE_3RD_POWER;
-            sum += inputCharDecoder.decodeInputChar(inputIndex + 2) * BASE_2ND_POWER;
-            sum += inputCharDecoder.decodeInputChar(inputIndex + 3) * BASE;
-            sum += inputCharDecoder.decodeInputChar(inputIndex + 4);
+            sum += decodeInputChar(input[inputIndex]) * BASE_4TH_POWER;
```

Comment:
`decodeInputChar` returns a long and `sum` is  an `int`.
which one is correct?

Reply:
`int` is the expected. Updated.

Reply:
`BASE*` constants are `long`, so the right-hand-side of compound assignment is `long`, but the variable is `int`.
in Java, this leads to silent truncation

let's declare `sum` as long and then add `toIntExact` in `buffer.putInt(sum);`

Reply:
It doesn't work. This logic is the same as Delta Lake code and it somehow depends on this int ↔︎ long.

Reply:
https://github.com/trinodb/trino/pull/18466

Reply:
How can this depend on numeric overflow?